rotkehlchen/api/rest.py

```diff
@@ -2695,20 +2694,28 @@ def delete_blockchain_transaction_data(
             with self.rotkehlchen.data.db.user_write() as write_cursor:
                 write_cursor.execute(querystr, bindings)
 
+        if not chain or chain == SupportedBlockchain.BITCOIN:
+            with self.rotkehlchen.data.db.user_write() as write_cursor:
+                self.rotkehlchen.data.db.delete_dynamic_caches(
+                    write_cursor=write_cursor,
+                    key_parts=[DBCacheDynamic.LAST_BITCOIN_TX_BLOCK.value[0].removesuffix('{address}')],
+                )
```

Comment:
can you split the different parts in functions and do something like

```python
    if not chain:
        # Case 1: No chain specified, delete data for all supported types.
        self._delete_evm_data(tx_hash=tx_hash)
        self._delete_zksync_data(tx_hash=tx_hash)
        self._delete_bitcoin_data()
    elif chain.is_evm():
        self._delete_evm_data(tx_hash=tx_hash)
    elif chain == SupportedBlockchain.ZKSYNC_LITE:
        self._delete_zksync_data(tx_hash=tx_hash)
    elif chain == SupportedBlockchain.BITCOIN:
        self._delete_bitcoin_data()
```

it helps to undersrtand better what is happening